src/overdub.jl

```diff
@@ -121,7 +121,7 @@ julia> ctx.metadata.x
 13.0
 ```
 """
-@inline posthook(::Context, ::Vararg{Any}) = nothing
+@inline posthook(::Context, output, ::Vararg{Any}) = output
 
 """
 ```
```

Comment:
(commenting here since github doesn't allow me to comment further down the file)

Additionally, commenting out the following methods fixes the miscompile (but, similar to the other "fix", causes miscompiles elsewhere):

```julia
@inline call(::ContextWithTag{Nothing}, f::typeof(Core.apply_type), ::Type{A}, ::Type{B}) where {A,B} = f(A, B)
@inline call(::Context, f::typeof(Core.apply_type), ::Type{A}, ::Type{B}) where {A,B} = f(A, B)
```

These methods are not actually called either way, but another method of `call` does seem to be the source of the miscompilation in the test added here.